src/core/message.ts

```diff
@@ -113,6 +113,13 @@ export abstract class Message {
     return currentNewestMessage;
   }
 
+  /**
+   * Gets the recordId of the given message.
+   */
+  public static getRecordId(message: BaseMessage): string {
+    return message['recordId'] ?? message.descriptor['recordId'];
```

Comment:
`recordId` is a  concept only to `Records` message, so to have a method for getting the `recordId` here is odd. Also if `message['recordId']` is `undefined, it doesn't mean `message.descriptor['recordId']` will be defined, maybe the reason for doing this is just a "best effort" to get the `recordId`?

I feel the alternatives below maybe "better" given the lack of options:
1. declare `recordId?: string` in `BaseMessage`, or if you feel this has practically the same issue
2. straight up doing `message['recordId']` in the places that invokes `getRecordId(...)` or
3. maybe express the true intent by having an abstract instanced method `getLogicalId()` where it is the `recordId` for `RecordsWrite`, and it's the `protocol` (URL) for `ProtocolConfigure`, but this change is more involved because you'll need to touch all interface methods, and logical ID does not apply to all methods (e.g. `RecordsQuery`) so in that case you'll probably have to return `undefined`... just more involved than you'd care for.

Just publishing my thought as I go (so that you don't get all comments in a big bang) and might revisit as I gain more context by reviewing further, I can already see by skimming through that maybe your idea of using `messageCid` is much better after all as it probably can sidestep all this discussion.

Reply:
yeah this was by best effort attempt at getting some sort of `recordId`

there isn't a `BaseRecord` class that `RecordsWrite`/`RecordsDelete`/etc. inherit from, so i felt this was the next best option (tho i could add that)

the reason im doing both is because `recordId` is defined on

- `RecordsWriteMessage` (_not_ `RecordsWriteDescriptor`)
- `RecordsDeleteDescriptor` (_not_ `RecordsDeleteMessage`)

so there's no consistency between them as to where it will be, and `StorageController`/`deleteAllOlderData`/etc. only know about `TimestampedMessage` so they have to handle both scenarios (btw this would be improved by [having actual `class` instead of manipulating the raw JSON objects](https://github.com/TBD54566975/dwn-sdk-js/issues/230#issuecomment-1435385854))